src/lib/jitter/jitter.cpp

```diff
@@ -0,0 +1,67 @@
+/*
+* Jitter utility functions
+* (C) 2024 Planck Security S.A.
+*
+* Botan is released under the Simplified BSD License (see license.txt)
+*/
+
+#include <span>
+
+#include "jitter.h"
+
+#ifdef BOTAN_HAS_JITTER
+
+   #include <memory>
+
+namespace Botan {
+
+rand_data* jitter_collector_create() {
+   static int result = jent_entropy_init();
+
+   if(result == 0) {
+      return jent_entropy_collector_alloc(1, 0);
+   } else {
+      return nullptr;
+   }
+}
+
+void jitter_collector_free(rand_data* collector) {
+   if(collector) {
+      jent_entropy_collector_free(collector);
+   }
+}
+
+void jitter_buffer(rand_data* collector, std::span<uint8_t> buf) {
+   jitter_buffer(collector, buf.data(), buf.size());
+}
+
+void jitter_buffer(rand_data* collector, uint8_t buf[], size_t len) {
+   if(!collector) {
+      return;
+   }
+
+   if(len <= 0) {
+      return;
+   }
+
+   std::unique_ptr<char[]> jitter_data{new char[len]};
```

Comment:
I don't understand why you would allocate this extra buffer; why not just write directly to buf?